diff -r -u -d org/src/declarativeimports/core/windowthumbnail.cpp fix/src/declarativeimports/core/windowthumbnail.cpp
--- org/src/declarativeimports/core/windowthumbnail.cpp	2026-06-23 16:13:25.000000000 +0800
+++ fix/src/declarativeimports/core/windowthumbnail.cpp	2026-06-30 07:28:39.637932788 +0800
@@ -6,7 +6,6 @@
 #include "windowthumbnail.h"
 // KF5
 #include <KWindowSystem>
-#include <KX11Extras>
 // Qt
 #include <QGuiApplication>
 #include <QIcon>
@@ -282,10 +281,6 @@
     if (m_winId == winId) {
         return;
     }
-    if (KWindowSystem::isPlatformX11() && !KX11Extras::self()->hasWId(winId)) {
-        // invalid Id, don't updated
-        return;
-    }
     if (window() && winId == window()->winId()) {
         // don't redirect to yourself
         return;
@@ -395,12 +390,8 @@
 void WindowThumbnail::iconToTexture(WindowTextureProvider *textureProvider)
 {
     QIcon icon;
-    if (KWindowSystem::isPlatformX11() && KX11Extras::self()->hasWId(m_winId)) {
-        icon = KX11Extras::self()->icon(m_winId, boundingRect().width(), boundingRect().height());
-    } else {
         // fallback to plasma icon
         icon = QIcon::fromTheme(QStringLiteral("plasma"));
-    }
     QImage image = icon.pixmap(boundingRect().size().toSize(), window()->devicePixelRatio()).toImage();
     textureProvider->setTexture(window()->createTextureFromImage(image, QQuickWindow::TextureCanUseAtlas));
 }
diff -r -u -d org/src/plasma/private/blureffectwatcher.cpp fix/src/plasma/private/blureffectwatcher.cpp
--- org/src/plasma/private/blureffectwatcher.cpp	2026-06-23 16:13:25.000000000 +0800
+++ fix/src/plasma/private/blureffectwatcher.cpp	2026-06-30 07:19:47.668762204 +0800
@@ -103,39 +103,14 @@
 {
 }
 
-#if HAVE_X11
 bool BlurEffectWatcher::nativeEventFilter(const QByteArray &eventType, void *message, qintptr *result)
 {
     if (KWindowSystem::isPlatformWayland()) {
         return false;
     }
-
-    Q_UNUSED(result);
-    // A faster comparison than eventType != "xcb_generic_event_t"
-    // given that eventType can only have the following values:
-    // "xcb_generic_event_t", "windows_generic_MSG" and "mac_generic_NSEvent"
-    // According to https://doc.qt.io/qt-5/qabstractnativeeventfilter.html
-    if (eventType[0] != 'x') {
-        return false;
-    }
-    xcb_generic_event_t *event = reinterpret_cast<xcb_generic_event_t *>(message);
-    uint response_type = event->response_type & ~0x80;
-    if (response_type != XCB_PROPERTY_NOTIFY || m_property == XCB_ATOM_NONE) {
-        return false;
-    }
-
-    xcb_property_notify_event_t *prop_event = reinterpret_cast<xcb_property_notify_event_t *>(event);
-    if (prop_event->atom == m_property) {
-        bool nowEffectActive = fetchEffectActive();
-        if (m_effectActive != nowEffectActive) {
-            m_effectActive = nowEffectActive;
-            Q_EMIT effectChanged(m_effectActive);
-        }
-    }
-
     return false;
 }
-
+#if HAVE_X11
 bool BlurEffectWatcher::fetchEffectActive() const
 {
     if (m_property == XCB_ATOM_NONE || !m_x11Interface) {
diff -r -u -d org/src/plasma/private/blureffectwatcher_p.h fix/src/plasma/private/blureffectwatcher_p.h
--- org/src/plasma/private/blureffectwatcher_p.h	2026-06-23 16:13:25.000000000 +0800
+++ fix/src/plasma/private/blureffectwatcher_p.h	2026-06-30 07:19:47.668762204 +0800
@@ -12,7 +12,7 @@
 
 #include <QAbstractNativeEventFilter>
 
-#include <xcb/xcb.h>
+
 
 #include <config-plasma.h>
 
@@ -30,9 +30,7 @@
     bool isEffectActive() const;
 
 protected:
-#if HAVE_X11
     bool nativeEventFilter(const QByteArray &eventType, void *message, qintptr *) override;
-#endif
 
 Q_SIGNALS:
     void effectChanged(bool on);
diff -r -u -d org/src/plasma/private/theme_p.cpp fix/src/plasma/private/theme_p.cpp
--- org/src/plasma/private/theme_p.cpp	2026-06-23 16:13:25.000000000 +0800
+++ fix/src/plasma/private/theme_p.cpp	2026-06-30 07:19:47.668762204 +0800
@@ -17,7 +17,6 @@
 #include <KSharedConfig>
 #include <KWindowEffects>
 #include <KWindowSystem>
-#include <KX11Extras>
 
 namespace Plasma
 {
@@ -92,9 +91,6 @@
     , apiMinor(0)
     , apiRevision(0)
 {
-    if (KWindowSystem::isPlatformX11()) {
-        compositingActive = KX11Extras::self()->compositingActive();
-    }
 
     kSvgImageSet = std::unique_ptr<KSvg::ImageSet>(new KSvg::ImageSet);
     kSvgImageSet->setBasePath(QStringLiteral(PLASMA_RELATIVE_DATA_INSTALL_DIR "/desktoptheme/"));
@@ -135,9 +131,6 @@
         scheduleThemeChangeNotification();
     });
 
-    if (KWindowSystem::isPlatformX11()) {
-        connect(KX11Extras::self(), &KX11Extras::compositingChanged, selectorsUpdateTimer, qOverload<>(&QTimer::start));
-    }
     updateKSvgSelectors();
 }
 
diff -r -u -d org/src/plasmaquick/appletpopup.cpp fix/src/plasmaquick/appletpopup.cpp
--- org/src/plasmaquick/appletpopup.cpp	2026-06-23 16:13:25.000000000 +0800
+++ fix/src/plasmaquick/appletpopup.cpp	2026-06-30 07:19:47.668762204 +0800
@@ -13,7 +13,6 @@
 
 #include <KConfigGroup>
 #include <KWindowSystem>
-#include <KX11Extras>
 
 #include "applet.h"
 #include "appletquickitem.h"
@@ -60,12 +59,9 @@
     setAnimated(true);
     setFlags(flags() | Qt::Dialog);
 
-    if (KWindowSystem::isPlatformX11()) {
-        KX11Extras::setType(winId(), NET::AppletPopup);
-    } else {
-        PlasmaShellWaylandIntegration::get(this)->setRole(QtWayland::org_kde_plasma_surface::role::role_appletpopup);
-        PlasmaShellWaylandIntegration::get(this)->setTakesFocus(true);
-    }
+    PlasmaShellWaylandIntegration::get(this)->setRole(QtWayland::org_kde_plasma_surface::role::role_appletpopup);
+    PlasmaShellWaylandIntegration::get(this)->setTakesFocus(true);
+
 
     auto edgeForwarder = new EdgeEventForwarder(this);
     edgeForwarder->setMargins(padding());
diff -r -u -d org/src/plasmaquick/dialog.cpp fix/src/plasmaquick/dialog.cpp
--- org/src/plasmaquick/dialog.cpp	2026-06-23 16:13:25.000000000 +0800
+++ fix/src/plasmaquick/dialog.cpp	2026-06-30 07:19:47.668762204 +0800
@@ -8,7 +8,6 @@
 */
 
 #include "dialog.h"
-#include "../declarativeimports/core/config-x11.h"
 #include "appletquickitem.h"
 #include "config-plasma.h"
 #include "configview.h"
@@ -24,9 +23,7 @@
 #include <QQuickItem>
 #include <QScreen>
 
-#include <KWindowInfo>
 #include <KWindowSystem>
-#include <KX11Extras>
 
 #include <KWindowEffects>
 #include <Plasma/Corona>
@@ -253,15 +250,10 @@
                                                  theme.backgroundSaturation(),
                                                  mask);
 
-        if (!KWindowSystem::isPlatformX11() || KX11Extras::compositingActive()) {
             if (hasMask) {
                 hasMask = false;
                 q->setMask(QRegion());
             }
-        } else {
-            hasMask = true;
-            q->setMask(dialogBackground->mask());
-        }
         if (q->isVisible()) {
             DialogShadows::instance()->addWindow(q, dialogBackground->enabledBorders());
         }
@@ -748,9 +740,6 @@
     }
 #endif
 
-    if (!wmType && type != Dialog::Normal && KWindowSystem::isPlatformX11()) {
-        KX11Extras::setType(q->winId(), static_cast<NET::WindowType>(type));
-    }
     if (q->flags() & Qt::WindowStaysOnTopHint) {
         // If the AppletPopup type is not explicitly requested, then use the Dock type in this case
         // to avoid bug #454635.
@@ -813,14 +802,6 @@
         }
     }
 
-    if (KWindowSystem::isPlatformX11()) {
-        if (type == Dialog::Dock || type == Dialog::Notification || type == Dialog::OnScreenDisplay || type == Dialog::CriticalNotification) {
-            KX11Extras::setOnAllDesktops(q->winId(), true);
-        } else {
-            KX11Extras::setOnAllDesktops(q->winId(), false);
-        }
-    }
-
     PlasmaShellWaylandIntegration::get(q)->setTakesFocus(!q->flags().testFlag(Qt::WindowDoesNotAcceptFocus));
 }
 
@@ -1079,13 +1060,7 @@
     }
 
     // if the item is in a window that ignores WM we want to position the popups outside
-    bool outsideParentWindow = (item->window()->flags() & Qt::X11BypassWindowManagerHint) && item->window()->mask().isNull();
-
-    if (KWindowSystem::isPlatformX11()) {
-        // on X11 we also consider windows with the type Dock
-        const KWindowInfo winInfo(item->window()->winId(), NET::WMWindowType);
-        outsideParentWindow = outsideParentWindow || (winInfo.windowType(NET::AllTypesMask) == NET::Dock && item->window()->mask().isNull());
-    }
+    bool outsideParentWindow = item->window()->mask().isNull();
 
     QRect parentGeometryBounds;
     if (outsideParentWindow) {
@@ -1361,9 +1336,6 @@
         DialogShadows::instance()->addWindow(this, d->dialogBackground->enabledBorders());
     }
 
-    if (KWindowSystem::isPlatformX11()) {
-        KX11Extras::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
-    }
     QQuickWindow::showEvent(event);
 }
 
diff -r -u -d org/src/plasmaquick/plasmawindow.cpp fix/src/plasmaquick/plasmawindow.cpp
--- org/src/plasmaquick/plasmawindow.cpp	2026-06-23 16:13:25.000000000 +0800
+++ fix/src/plasmaquick/plasmawindow.cpp	2026-06-30 07:19:47.668762204 +0800
@@ -13,7 +13,6 @@
 
 #include <KWindowEffects>
 #include <KWindowSystem>
-#include <KX11Extras>
 
 #include <Plasma/Theme>
 
@@ -143,9 +142,6 @@
 {
     // EWMH states that the state is reset every hide
     // Qt supports external factors setting state before the next show
-    if (KWindowSystem::isPlatformX11()) {
-        KX11Extras::setState(winId(), NET::SkipTaskbar | NET::SkipPager | NET::SkipSwitcher);
-    }
     QQuickWindow::showEvent(e);
 }
 
@@ -171,11 +167,7 @@
                                              theme.backgroundSaturation(),
                                              mask);
 
-    if (!KWindowSystem::isPlatformX11() || KX11Extras::compositingActive()) {
-        q->setMask(QRegion());
-    } else {
         q->setMask(mask);
-    }
 }
 
 void PlasmaWindowPrivate::updateMainItemGeometry()
diff -r -u -d org/src/plasmaquick/popupplasmawindow.cpp fix/src/plasmaquick/popupplasmawindow.cpp
--- org/src/plasmaquick/popupplasmawindow.cpp	2026-06-23 16:13:25.000000000 +0800
+++ fix/src/plasmaquick/popupplasmawindow.cpp	2026-06-30 07:19:47.668762204 +0800
@@ -31,7 +31,6 @@
     void updateEffectivePopupDirection(const QRect &anchorRect, const QRect &relativePopupPosition);
     void updateSlideEffect(const QRect &globalPosition);
     void updatePosition();
-    void updatePositionX11(const QPoint &position);
     void updatePositionWayland(const QPoint &position);
     void updateBorders(const QRect &globalPosition);
     void updateVisualParentWindow();
@@ -177,20 +176,13 @@
     updateEffectivePopupDirection(parentAnchorRect.toRect(), relativePopupPosition);
     updateSlideEffect(popupPosition);
 
-    if (KWindowSystem::isPlatformX11()) {
-        updatePositionX11(popupPosition.topLeft());
-    } else if (KWindowSystem::isPlatformWayland()) {
+    if (KWindowSystem::isPlatformWayland()) {
         updatePositionWayland(popupPosition.topLeft());
     }
 
     updateBorders(popupPosition);
 }
 
-void PopupPlasmaWindowPrivate::updatePositionX11(const QPoint &position)
-{
-    q->setPosition(position);
-}
-
 void PopupPlasmaWindowPrivate::updatePositionWayland(const QPoint &position)
 {
     // still update's Qt internal reference as it's used by the next dialog
